tests/test_api.rs

```diff
@@ -1553,6 +1563,29 @@ fn object() {
     assert!(!object.has(scope, n_unused).unwrap());
     assert!(object.delete(scope, n1.into()).unwrap());
     assert!(!object.has(scope, n1.into()).unwrap());
+
+    let global = context.global(scope);
+    let object_string = v8::String::new(scope, "object").unwrap().into();
+    global.set(scope, object_string, object.into());
+    assert!(object
+      .set_integrity_level(scope, v8::IntegrityLevel::Sealed)
+      .unwrap());
+    {
+      let scope = &mut v8::TryCatch::new(scope);
+      // Creating a new property fails (in strict mode).
+      eval_strict(scope, "object.c = 'foo';");
+      assert!(scope.has_caught());
```

Comment:
In non-strict mode, this silently does nothing.

Reply:
It doesn't throw if that is what you mean but it's still observable because `object.c` won't have been set.

The way you can test this in sloppy mode is by checking that `Object.isSealed()` and `Object.isExtensible()` return false and true before and the inverse afterward - and that the property hasn't been set, of course. Ditto for frozen mode.

Aside: I'd prefer not to have this `eval_strict()` helper. It obscures more than it clarifies, IMO.

Reply:
>The way you can test this in sloppy mode is by checking that Object.isSealed() and Object.isExtensible() return false and true before and the inverse afterward - and that the property hasn't been set, of course. Ditto for frozen mode.

I did not think of `Object.isSealed`, `isExtensible` and `isFrozen`... I'm actually tempted to "trust" their return value, and so to skip trying to create/mutate a property and observing the expected side-effect (or absence thereof). I'm leaving those in for now, though...

> Aside: I'd prefer not to have this `eval_strict()` helper. It obscures more than it clarifies, IMO.

Yeah... Since this can be tested fully in sloppy mode, it does not make a ton of sense. I initially added it largely because I found `'use strict';\nobject.c = 'foo';` looked kinda bad...

Reply:
You can use multi-line strings for that, they also let you embed quotes:
```rust
let source = r#"
  "use strict";
  object.c = "foo";
"#;
```